hydra-check: 2.0.1 -> 2.0.3

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar committed:
hydra-check: 2.0.1 -> 2.0.2

Diff: https://github.com/nix-community/hydra-check/compare/v2.0.1...v2.0.2

SuperSandro2000 commented on pkgs/by-name/hy/hydra-check/package.nix:
```diff
@@ -10,16 +10,16 @@
   src = fetchFromGitHub {
     owner = "nix-community";
     repo = "hydra-check";
     rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

bryango reviewed (approved):
While we are at it, might as well add a `versionCheckHook`? :laughing: e.g.
```patch
From 289d5c708648a63b19d376ebdfb7b2bcf07e2fff Mon Sep 17 00:00:00 2001
From: Bryan Lai <bryanlais@gmail.com>
Date: Sun, 2 Feb 2025 15:34:03 +0800
Subject: [PATCH] hydra-check: add versionCheckHook & doInstallCheck

See: https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook
---
 pkgs/by-name/hy/hydra-check/package.nix | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pkgs/by-name/hy/hydra-check/package.nix b/pkgs/by-name/hy/hydra-check/package.nix
index 46823dab7d69..dd1e363c7f1a 100644
--- a/pkgs/by-name/hy/hydra-check/package.nix
+++ b/pkgs/by-name/hy/hydra-check/package.nix
@@ -6,6 +6,7 @@
   openssl,
   stdenv,
   installShellFiles,
+  versionCheckHook,
 }:
 
 rustPlatform.buildRustPackage rec {
@@ -37,6 +38,12 @@ rustPlatform.buildRustPackage rec {
       --zsh <($out/bin/hydra-check --shell-completion zsh)
   '';
 
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+
+  doInstallCheck = true;
+
   meta = {
     description = "Check hydra for the build status of a package";
     homepage = "https://github.com/nix-community/hydra-check";
```
This is implemented [upstream](https://github.com/nix-community/hydra-check/blob/85de859f5c93a6ac9392c4563d7dcec13bedf4df/package.nix#L29-L35) and it makes sure that we do use the correct git source & hash. I often mess that up :laughing:

doronbehar committed:
hydra-check: add versionCheckHook

doronbehar commented:
Haha I added `versionCheckHook` and it did not pass!

bryango commented:
Oh shoot I messed up again... Forgot to bump the version number in `Cargo.toml`... Fixing it in https://github.com/nix-community/hydra-check/pull/64, will merge and release `v2.0.3` once CI is green.

doronbehar committed:
hydra-check: 2.0.2 -> 2.0.3

Diff: https://github.com/nix-community/hydra-check/compare/refs/tags/v2.0.2...v2.0.3

bryango reviewed (approved):
Tested build on `x86_64-linux`, looking good with or without `useFetchCargoVendor`. Thanks!

doronbehar commented on pkgs/by-name/hy/hydra-check/package.nix:
```diff
@@ -6,20 +6,21 @@
   };
 
-  cargoHash = "sha256-iqFUMok36G1qSUbfY7WD6etY0dtfro3F7mLoOELzxbs=";
+  cargoHash = "sha256-AUIm8OLN+BgEtlYCFQkozEwkp0Iwfkj+l5ttSAoCz60=";
```
> Ah I just realized, maybe we should `useFetchCargoVendor`?
> 
> I hope I get the hash right this time 😿 Of course this is non-blocking! Builds fine either way, approved. Thank you very much!

TBH I didn't understand yet the point of that transition to `useFetchCargoVendor` which I have seen many PRs for. I thought this was needed only for certain packages, but I'm not sure what are the criteria .. maybe @alyssais could help us decide?

alyssais replied:
It is needed for all packages that are not using importCargoLock. See #378288.

doronbehar replied:
OK done. Thanks @alyssais . Will merge when CI is green.

doronbehar committed:
hydra-check: useFetchCargoVendor

See #378288
